### PR TITLE
feat: add SonarQube configurations

### PR DESCRIPTION
**Pull request by apurv-wednesday**
* Add `jest-sonar` plugin for integrating Jest with SonarQube
* Add SonarQube project properties to configure analysis settings
* Add SonarQube checks on CI pipeline

## Summary by CodeRabbit

- **Chores**
	- Temporarily disabled Lighthouse CI to maintain a green CI pipeline.
	- Integrated SonarQube for enhanced code quality analysis.
- **New Features**
	- Added `jest-sonar` for improved test reporting in XML format.
- **Documentation**
	- Configured SonarQube analysis settings for the project.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,12 +29,17 @@ jobs:
...
-      #     yarn global add @lhci/cli@0.8.x
-      #     lhci autorun --collect.staticDistDir
-
+      
+        # commenting for now to make CI green , will try this 
+        # - name: run Lighthouse CI
+        #   run: |
+        #     yarn global add @lhci/cli@0.8.x
+        #     lhci autorun --collect.staticDistDir
+
```
[the remark this answers is not in the document]
Comment (apurv-wednesday) [reply .github/workflows/ci.yml:38]: yes

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,12 +29,17 @@ jobs:
...
       - name: Build Project
         run: yarn build
-
-      # commenting for now to make CI green , will try this 
-      # - name: run Lighthouse CI
-      #   run: |
-      #     yarn global add @lhci/cli@0.8.x
-      #     lhci autorun --collect.staticDistDir
-
+
```
Comment (alichherawalla) [inline .github/workflows/ci.yml:32]: in a subsequent PR let's integrate support for this

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,11 @@
+sonar.organization=wednesday-solutions
+sonar.projectKey=wednesday-solutions_next-bulletproof-ts_AY6yu6eKB2n8RRmGoUz4
+
+sonar.language=ts
+sonar.sources=.
+sonar.tests=app
+sonar.exclusions=**/tests/**/*.*,**/storybook/**/*.*,**/internals/**/*.*,**/server/**/*.*,app/**/*.test.{ts,tsx},app/*/RbGenerated*/*.{ts,tsx},app/app.ts,app/global-styles.ts,app/*/*/loadable.{ts,tsx},**/loadable.ts,**/apiUtils.ts,**/testUtils.ts,**/stories/**,**/themes/index.ts
```
Comment (alichherawalla) [inline sonar-project.properties:7]: are all these still applicable for this project?

**Comment by alichherawalla** [comment]
before we merge this can you check if the results are actually getting sent to sonar correctly. You may need to change the trigger temporarily to solve for it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (apurv-wednesday) [reply .github/workflows/ci.yml:32]: Sure

File: sonar-project.properties (hunk 3 above)
Comment (apurv-wednesday) [reply sonar-project.properties:7]: will remove it for now

Comment (alichherawalla) [reply sonar-project.properties:7]: let's note remove it, lets figure what is applicable and add only those

**Pull request closed without merging**